src/language-server/config/config.ts

```diff
@@ -1,146 +1,240 @@
 import { dirname } from "path";
-import merge from "lodash.merge";
-import { ClientID, ServiceID, ServiceSpecifier } from "../engine";
 import { URI } from "vscode-uri";
-import { WithRequired } from "../../env";
 import { getGraphIdFromConfig, parseServiceSpecifier } from "./utils";
+import { Debug } from "../utilities";
+import z, { ZodError } from "zod";
```

Comment:
Moving everything over to zod. GitHub might collapse the file, but this is where the magic happens.